src/Catalyst.Cli/CommandTypes/BaseCommand.cs

```diff
@@ -0,0 +1,68 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Reflection;
+using Catalyst.Abstractions.Cli.Commands;
+using Catalyst.Abstractions.Cli.CommandTypes;
+using Catalyst.Abstractions.Cli.Options;
+using CommandLine;
+using Serilog;
+
+namespace Catalyst.Cli.CommandTypes
+{
+    public abstract class BaseCommand<TOption> : ICommand
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Convert this 'abstract' class to a concrete class with a protected constructor.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)